packages/editor-ui/src/stores/ui.store.ts

```diff
@@ -87,545 +81,660 @@ try {
-								tooltip: `contextual.workflows.sharing.unavailable.description.tooltip${contextKey}`,
-							},
-							action: `contextual.workflows.sharing.unavailable.action${contextKey}`,
-							button: `contextual.workflows.sharing.unavailable.button${contextKey}`,
-						},
-					},
+			},
+			variables: {
+				unavailable: {
+					title: `contextual.variables.unavailable.title${contextKey}`,
+					description: 'contextual.variables.unavailable.description',
+					action: `contextual.variables.unavailable.action${contextKey}`,
+					button: `contextual.variables.unavailable.button${contextKey}`,
 				},
-				variables: {
+			},
+			users: {
+				settings: {
 					unavailable: {
-						title: `contextual.variables.unavailable.title${contextKey}`,
-						description: 'contextual.variables.unavailable.description',
-						action: `contextual.variables.unavailable.action${contextKey}`,
-						button: `contextual.variables.unavailable.button${contextKey}`,
+						title: `contextual.users.settings.unavailable.title${contextKey}`,
+						description: `contextual.users.settings.unavailable.description${contextKey}`,
+						button: `contextual.users.settings.unavailable.button${contextKey}`,
 					},
 				},
-				users: {
-					settings: {
-						unavailable: {
-							title: `contextual.users.settings.unavailable.title${contextKey}`,
-							description: `contextual.users.settings.unavailable.description${contextKey}`,
-							button: `contextual.users.settings.unavailable.button${contextKey}`,
-						},
-					},
-				},
-			} as const;
-		},
-		getLastSelectedNode(): INodeUi | null {
-			const workflowsStore = useWorkflowsStore();
-			if (this.lastSelectedNode) {
-				return workflowsStore.getNodeByName(this.lastSelectedNode);
+			},
+		} as const;
+	});
+
+	const getLastSelectedNode = computed(() => {
+		if (lastSelectedNode.value) {
+			return workflowsStore.getNodeByName(lastSelectedNode.value);
+		}
+		return null;
+	});
+
+	const isVersionsOpen = computed(() => {
+		return modals.value[VERSIONS_MODAL_KEY].open;
+	});
+
+	const isModalOpen = computed(() => {
+		return (name: ModalKey) => modals.value[name].open;
+	});
+
+	const isModalActive = computed(() => {
+		return (name: ModalKey) => modalStack.value.length > 0 && name === modalStack.value[0];
+	});
+
+	const getModalActiveId = computed(() => {
+		return (name: ModalKey) => modals.value[name].activeId;
+	});
+
+	const getModalMode = computed(() => {
+		return (name: ModalKey) => modals.value[name].mode;
+	});
+
+	const getModalData = computed(() => {
+		return (name: ModalKey) => modals.value[name].data;
+	});
+
+	const getFakeDoorByLocation = computed(() => {
+		return (location: IFakeDoorLocation) =>
+			fakeDoorFeatures.value.filter((fakeDoor) => fakeDoor.uiLocations.includes(location));
+	});
+
+	const getFakeDoorById = computed(() => {
+		return (id: string) => fakeDoorFeatures.value.find((fakeDoor) => fakeDoor.id.toString() === id);
+	});
```

Comment:
Same comment as above.

I know it's a lot to ask, but we'll need to do this for the other stores as well.

Reply:
All good! I will do it for this one and for the other stores as well!